src/main/java/io/cdap/plugin/gcp/publisher/source/GoogleSubscriber.java

```diff
@@ -149,5 +152,36 @@ private ZonedDateTime getTimestamp(String publishTime) {
     @Macro
     @Nullable
     private String topic;
+
+    public void validate(FailureCollector collector) {
+      super.validate(collector);
+      String regAllowedChars = "[A-Za-z0-9-.%~+_]*$";
+      String regStartWithLetter = "[A-Za-z]";
+      if (!getSubscription().matches(regAllowedChars)) {
+        collector.addFailure("Subscription Name does not match naming convention.",
+          "Unexpected Character. Check Plugin documentation for naming convention")
```

Comment:
nit: period at the end of sentence here and in other error messages.

Reply:
completed